rego/rego.go

```diff
@@ -930,9 +941,11 @@ func (r *Rego) PartialResult(ctx context.Context) (PartialResult, error) {
 	}
 
 	pr := PartialResult{
-		compiler: pq.r.compiler,
-		store:    pq.r.store,
-		body:     pq.r.parsedQuery,
+		compiler:     pq.r.compiler,
+		store:        pq.r.store,
+		body:         pq.r.parsedQuery,
+		builtinDecls: pq.r.builtinDecls,
+		builtinFuncs: pq.r.builtinFuncs,
 	}
```

Comment:
I'm curious why we need these two separate code paths for constructing the PartialResult objects. Would it be better to have one? This only came to mind while reviewing the PR. It doesn't have to be addressed immediately (unless you think it's an easy fix.)

Reply:
As-is what happens is API callers can do `PrepareForEval` and specify `WithPartialEval`, which will do the partial evaluation, get a Rego object from it, and prepare that one (the returned PreparedEvalQuery). We re-use that some code path with the older `PartialResult` API which takes the prepared Rego object (from that previous flow) and makes a `PartialResult` out of it.

So.. do we need to do that? No, probably not. We could just have the first PartialResult object get returned on the `pq` and return it without doing a prepare step on its `Rego` object.